src/main/java/Duke.java

```diff
@@ -1,10 +1,261 @@
+                        if (s1.equals("")) {
+                            throw new DukeException("Enter a valid todo");
+                        } else {
+                            char[] chars = s1.toCharArray();
+                            if(chars[1] == ' '){
+                                throw new DukeException("Enter valid todo");
+                            } else {
+                                try{
+                                    System.out.println("Got it. I've added this todo:");
+                                    String desc = s1.substring(1);
+                                    Todo newTodo = new Todo(desc);
+                                    System.out.println(" " + newTodo.toString());
+                                    myList.add(newTodo);
+                                    System.out.println("Now you have " + myList.size() + " tasks in the list.");
+                                    database.writeTaskToFile(myList);
+                                } catch (Exception e){
+                                    System.out.println("Enter valid todo");
+                                }
+
+                            }
+                        }
+                    } catch (DukeException e){
+                        System.out.println(e.getMessage());
+                    }
+                    System.out.println("____________________________________");
+                    break;
+
+                case "deadline":
+
+                    System.out.println("____________________________________");
+                    try{
+                        String s1 = input.nextLine();
+                        if(s1.equals("")){
+                            throw new DukeException("Enter valid deadline task.");
+                        } else {
+                            try {
+                                String wholeString = s1.substring(1);
+                                String[] parts = wholeString.split(" /by ");
+                                String deadlineDesc = parts[0];
+                                if(parts.length == 1){
+                                    throw new DukeException("Please adhere to convention:\n(task /by deadline timing)");
+                                } else {
+                                    String deadline = parts[1];
+                                    System.out.println("Got it. I've added this deadline:");
+                                    Deadline newDeadline = new Deadline(deadlineDesc, deadline);
+                                    System.out.println(" " + newDeadline.toString());
+                                    myList.add(newDeadline);
+                                    System.out.println("Now you have " + myList.size() + " tasks in the list.");
+                                    database.writeTaskToFile(myList);
+
+                                }
+                            } catch(DukeException e) {
```

Comment:
Not a coding standard, but figured I could chip in. Since the `catch` block is the same as that below in line 86, you can consider removing the `try-except` block here.